lib/backend_classes/filenames.dart

```diff
@@ -2,94 +2,63 @@ import 'package:daily_diary/main.dart';
 
 class Filename {
   static String dateToFilename(DateTime date, {String? dateFormat}) {
-    String filename = dateFormat ?? App.settingsNotifier.value.dateFormat;
-    filename = filename.replaceAll('%Y', _twoDigits(date.year));
-    filename = filename.replaceAll('%M', _twoDigits(date.month));
-    filename = filename.replaceAll('%D', _twoDigits(date.day));
-    return filename;
+    final format = dateFormat ?? App.settingsNotifier.value.dateFormat;
+    return format
+        .replaceAll('%Y', _twoDigits(date.year)) 
+        .replaceAll('%M', _twoDigits(date.month)) 
+        .replaceAll('%D', _twoDigits(date.day)); 
   }
 
-  static String _twoDigits(int n) {
-    if (n >= 10) return '$n';
-    return '0$n';
-  }
+ 
+  static String _twoDigits(int n) => n >= 10 ? '$n' : '0$n';
 
   static DateTime? filenameToDate(String filename) {
-    String dateFormat = App.settingsNotifier.value.dateFormat;
-    _Order order = _Order(dateFormat);
+    final dateFormat = App.settingsNotifier.value.dateFormat;
+    final order = _Order.fromFormat(dateFormat);
 
-    String regexString = dateFormat
-        .replaceFirst('%Y', r'(\d+)')
-        .replaceFirst('%M', r'(\d+)')
-        .replaceFirst('%D', r'(\d+)');
-    RegExp regex = RegExp(regexString);
+    final regexPattern = dateFormat
+        .replaceAll('%Y', r'(\\d{4})') 
+        .replaceAll('%M', r'(\\d{2})') 
+        .replaceAll('%D', r'(\\d{2})'); 
```

Comment:
Thank you for the PR. Although, the double backslash `\\` in the regex matches a literal `\` and does not work with the files, so I've reverted it back to a single backslash.
```suggestion
        .replaceAll('%Y', r'(\d{4})') 
        .replaceAll('%M', r'(\d{2})') 
        .replaceAll('%D', r'(\d{2})'); 
```